internal/pkg/build/build.go

```diff
@@ -332,8 +353,112 @@ func makeDef(spec string, remote bool) (types.Definition, error) {
 
 	d, err := parser.ParseDefinitionFile(defFile)
 	if err != nil {
-		return types.Definition{}, fmt.Errorf("While parsing definition: %s: %v", spec, err)
+		return types.Definition{}, fmt.Errorf("while parsing definition: %s: %v", spec, err)
+	}
+
+	return d, nil
+}
+
+// MakeAllDefs gets a definition slice from a spec
+func MakeAllDefs(spec string, remote bool) ([]types.Definition, error) {
+	return makeAllDefs(spec, remote)
+}
+
+// makeAllDef gets a definition object from a spec
+func makeAllDefs(spec string, remote bool) ([]types.Definition, error) {
+	if ok, err := uri.IsValid(spec); ok && err == nil {
+		// URI passed as spec
+		d, err := types.NewDefinitionFromURI(spec)
+		return []types.Definition{d}, err
+	}
+
+	// Check if spec is an image/sandbox
+	if _, err := image.Init(spec, false); err == nil {
+		d, err := types.NewDefinitionFromURI("localimage" + "://" + spec)
+		return []types.Definition{d}, err
+	}
+
+	// default to reading file as definition
+	defFile, err := os.Open(spec)
+	if err != nil {
+		return nil, fmt.Errorf("unable to open file %s: %v", spec, err)
+	}
+	defer defFile.Close()
+
+	// must be root to build from a definition
+	if os.Getuid() != 0 && !remote {
+		sylog.Fatalf("You must be the root user to build from a Singularity recipe file")
+	}
+
+	d, err := parser.All(defFile)
+	if err != nil {
+		return nil, fmt.Errorf("while parsing definition: %s: %v", spec, err)
 	}
 
 	return d, nil
 }
+
+func (b *Build) findStageIndex(name string) (int, error) {
+	for i, s := range b.stages {
+		if name == s.name {
+			return i, nil
+		}
+	}
+
+	return -1, fmt.Errorf("stage %s was not found", name)
+}
+
+func (s *stage) copyFiles(b *Build) error {
+	def := s.b.Recipe
+	for _, f := range def.BuildData.Files {
+		if f.Args == "" {
+			continue
+		}
+		args := strings.Fields(f.Args)
+		if len(args) != 2 {
+			continue
+		}
+
+		stageIndex, err := b.findStageIndex(args[1])
+		if err != nil {
+			return err
+		}
+
+		sylog.Debugf("Copying files from stage: %s", args[1])
+
+		// iterate through filetransfers
+		for _, transfer := range f.Files {
+			// sanity
+			if transfer.Src == "" {
+				sylog.Warningf("Attempt to copy file with no name...")
```

Comment:
Should this be the same as the other message?

```suggestion
				sylog.Warningf("Attempt to copy file with no name, skipping.")
```